nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -267,26 +267,36 @@ in
         mkdir -p /var/lib/xen # so we create them here unconditionally.
         grep -q control_d /proc/xen/capabilities
         '';
-      serviceConfig.ExecStart = ''
-        ${cfg.stored}${optionalString cfg.trace " -T /var/log/xen/xenstored-trace.log"} --no-fork
-        '';
+      serviceConfig = if cfg.package.version < "4.8" then
+	{ ExecStart = ''
+	    ${cfg.stored}${optionalString cfg.trace " -T /var/log/xen/xenstored-trace.log"} --no-fork
```

Comment:
Looking at the rest of the module, `/var/log/xen` seems to be hardcoded in a lot of other places. If you want to get rid of it, then I suggest to do it with a separate PR.